pkg/clusteragent/clusterchecks/dispatcher_configs.go

```diff
@@ -90,29 +90,27 @@ func (d *dispatcher) removeConfig(digest string) {
 
 	node, found := d.store.getNodeStore(d.store.digestToNode[digest])
 
-	checkName := ""
-	if found {
-		node.RLock()
-		checkName = node.digestToConfig[digest].Name
-		node.RUnlock()
-	}
+	checkName := d.store.digestToConfig[digest].Name
 
 	delete(d.store.digestToNode, digest)
 	delete(d.store.digestToConfig, digest)
 	delete(d.store.danglingConfigs, digest)
 
-	for k, v := range d.store.idToDigest {
-		if v == digest {
-			configsInfo.Delete(node.name, checkName, string(k), le.JoinLeaderValue)
-			delete(d.store.idToDigest, k)
+	var checkIDToRemove checkid.ID
+	for checkID, checkDigest := range d.store.idToDigest {
+		if checkDigest == digest {
+			checkIDToRemove = checkID
+			delete(d.store.idToDigest, checkIDToRemove)
 		}
 	}
```

Comment:
If there’s only one element in `d.store.idToDigest` for which the value is `digest`, we can break the for loop as soon as we’ve found it.

```suggestion
	var checkIDToRemove checkid.ID
	for checkID, checkDigest := range d.store.idToDigest {
		if checkDigest == digest {
			checkIDToRemove = checkID
			delete(d.store.idToDigest, checkIDToRemove)
			break
		}
	}
```

But if it’s possible to have several matches, `checkIDToRemove` should be a list.

Reply:
At first I thought there could only be one match, but I was wrong.
In configs with multiple instances each one has a check ID, so this needs to be a list.
I've pushed the fix :+1: